include/status-code/status_code.hpp

```diff
@@ -379,8 +379,20 @@ namespace detail
     {
     }
   };
+
+  template <> class status_code_storage<void>
+  {
+  };
 }  // namespace detail
 
+namespace mixins
+{
+  template <> struct mixin<detail::status_code_storage<void>, void> final
```

Comment:
I'm kinda thinking it's time for specific traits right? Would you like me to rework this PR for those?

Reply:
Adding a trait/extending `type_erasure_is_safe` is probably cleaner and would also reduces code duplication. If you outlined the preferred solution, I wouldn't mind trying it myself.

Reply:
There is already a `traits` subnamespace, so I guess all that duplicated logic currently in the status code and errored status code constructors if it could be lifted out into specific named traits that would be valuable. Normally I'd use templated constexpr variables here, but status code is a C++ 11 codebase, so you'll have to stick with types which quack like `std::true_type`.

Good traits design is a useful skill to master, so sure rock on giving it a try. To do it well, think like a compiler: compilers will calculate and cache the results of individual compile time expressions into a hash table for a given set of constant inputs. So, try to break out traits into subtraits which have least possible template parameters, and therefore most likely to already be in a hash table and therefore not need recalculating, and feel free to using a `traits::detail` namespace if needed for subtraits.

Equally, every time you tell the compiler about a subtrait you consume more memory and increase the size of structures which must be searched for a cached entry, so there is a tradeoff there between granulairty of subtraits and likelihood of hitting on a cached result. Sometimes it can make sense to string together long boolean logic clauses if breaking them into subtraits would likely be a negative impact rather than positive.

I don't claim excellence at traits design, but those are the boundaries to design within. Good luck!